sdk/lusid/models/weighted_instrument.py

```diff
@@ -44,24 +45,11 @@ class WeightedInstrument(object):
         'instrument': 'instrument'
     }
 
-    required_map = {
-        'quantity': 'optional',
-        'holding_identifier': 'optional',
-        'instrument': 'optional'
-    }
```

Comment:
What happened to required_map? This dictionary is critical as it allows other toolsets which consume the SDK such as LUSID Python Tools to know which fields are required and which are not. Without this there is no way to easily know which fields are required and which are optional. 

Looking at the merge request for the generator (https://github.com/finbourne/lusid-sdk-generators/pull/40/files) it looks as though the mustache template for the Python SDK models has gone missing completely? This is where the logic for generating this comes from which is as below:



<!--StartFragment-->    required_map = {{{#vars}}        '{{name}}': '{{^required}}optional{{/required}}{{#required}}required{{/required}}'{{#hasMore}},{{/hasMore}}{{/vars}}    }<!--EndFragment-->

Reply:
This is where we use this in LUSID Python Tools - https://github.com/finbourne/lusid-python-tools/blob/feddeb777cabc13fc92a28fbbe7b76f257832f04/lusidtools/cocoon/utilities.py#L239

Reply:
Hopefully I've fixed that now. 

There seem to be some massive changes in how code generation works if you use the `python` vs `python-legacy` types. Ideally we'd use the former, but it generates all the model codes under `sdk/model` instead of `sdk/models` and as far as I can tell, that's hardcoded in the generator - https://github.com/OpenAPITools/openapi-generator/blob/60dcf8613f6b75b606757c4a991fe018ad10ee99/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java#L126

It used to go to `models` as per https://github.com/OpenAPITools/openapi-generator/blob/3bbaedd9bc086028e7b1be973a30e04870962e5d/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java#L90 but if we want it to go there now we need to use the legacy generator instead.

Reply:
Okay I see that the `required_map` is back now, nice work! What is the significance of `model` vs `models` in relation to the `required_map`?

Reply:
Let's add this into the new template instead of reverting to the old template

Reply:
Okay, we have reverted to the old template and the legacy generator, nice!